iothub/service/src/Common/Exceptions/DeviceMessageLockLostException.cs

```diff
@@ -7,10 +7,9 @@
 namespace Microsoft.Azure.Devices.Common.Exceptions
 {
     /// <summary>
-    /// The exception that is thrown when an attempt to communicate with a device fails
-    /// because the lock token was lost (if the connection is lost and regained for example).
-    /// This timeout has the same effect as if the message was abandoned.
+    /// This exception does not appear to be returned by the service for ServiceClient operations.
     /// </summary>
+    [Obsolete("This exception does not appear to be returned by the service.")]
```

Comment:
Add `[EditorBrowsable(EditorBrowsableState.Never)]`?

Reply:
On second thought, the SDK still throws this exception (to avoid breaking change), so I don't think we should mark it as obsolete.

Reply:
Based on my analysis, this exception should not be thrown by the ServiceClient. However, there is no good way to confirm. There is no explanation I can add for when this exception would be thrown. Thoughts on how to proceed other than marking it as obsolete?

Reply:
Actually on further analysis, I think we are converting HttpStatus PreconditionFailed to this error for some reason. I will revert this change and just document that for this exception.